### DISCUSSION TITLE
Index out of range error when scanning a Windows 10 Enterprise target via WinRM

### DISCUSSION THREAD

**Discussion by djtek1234** [mondoohq/cnquery#3234] [Q&A]
Using cnspec 10.1.6. Trying to do a scan using the standard Windows policy of another target machine on the network. WinRM is enabled, basic auth is configured, unencrypted is allowed, and port 5985 is open. 

Command: "cnspec.exe scan winrm user@192.168.39.9:5985 --ask-pass -v"

Output: 

![image](https://github.com/mondoohq/cnquery/assets/12529410/af321d74-f2fc-4585-8cc1-39fbf2da3873)

I have had success scanning a different Windows target and noticed that there was a "platform_id" field shown that the above target does not seem to have. I tried setting one manually with the "--platform-id" flag with no luck. Perhaps I did not use that flag properly.

One last thing to note. I can connect to the above target if I do "cnspec.exe shell winrm ...". That tells me that this is not a WinRM issue.

**Comment by chris-rock** [comment]
@djtek1234 Thank you for the report. Can you run the following in powershell

```
$env:DEBUG=1
cnspec.exe scan winrm user@192.168.39.9:5985 --ask-pass -v
```

I think you are right, that the platform id was not properly detected in https://github.com/mondoohq/cnspec/blob/v10.2.0/policy/scan/local_scanner.go#L555. This should not happen.

To figure out why that is, can you run the following commands on the windows machine:

```
hostname
```

**Comment by djtek1234** [comment]
@chris-rock Thank you for the speedy reply. I ran cnspec within Powershell with the commands you supplied. The results are below. Apologies for using screen grabs, I do not have copy-paste setup on this VM. 

![image](https://github.com/mondoohq/cnquery/assets/12529410/5b137dea-bae8-4d1f-9b6d-4576e2c46430)
![image](https://github.com/mondoohq/cnquery/assets/12529410/7b86a702-d5a0-4302-b985-7249da5c741e)
![image](https://github.com/mondoohq/cnquery/assets/12529410/8490149e-5216-438e-bd0f-80abf352a629)
![image](https://github.com/mondoohq/cnquery/assets/12529410/65595f16-862d-4de5-a006-ef5c5e2f125b)
![image](https://github.com/mondoohq/cnquery/assets/12529410/d0ce81c0-ce3d-487a-acb8-fdacfc4331da)

The result for the hostname command on the WinRM machine is "DESKTOP-ABC" (I renamed this from DESKTOP- followed by random numbers and letters to see if things would change).

**Reply by chris-rock** [reply]
That is an iteressting one. clearly the hostname command returns an empty string, hence the empty asset name. We need to identify why hostname is return an empty result. Can you help and share the results of the following mql queries?

```
cnspec> os.hostname
os.hostname: "chris.fritz.box"
cnspec> asset.ids
asset.ids: [
  0: "//platformid.api.mondoo.app/hostname/chris.fritz.box"
]
```

**Comment by djtek1234** [comment]
![image](https://github.com/mondoohq/cnquery/assets/12529410/0cc4c393-af9b-431e-828e-6bc2e26afba8)

**Reply by chris-rock** [reply]
Ok, are you on latest cnspec? The output of `cnspec status` would be interesting.

**Reply by djtek1234** [reply]
![image](https://github.com/mondoohq/cnquery/assets/12529410/bfe817b5-a4d7-439b-a4f8-72523d8dc270)
